flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceOperatorStreamTaskTest.java

```diff
@@ -99,6 +102,29 @@ public void testSnapshotAndAdvanceToEndOfEventTime() throws Exception {
 		}
 	}
 
+	@Test
+	public void testEmittingMaxWatermarkAfterReadingAllRecords() throws Exception {
+		try (StreamTaskMailboxTestHarness<Integer> testHarness = createTestHarness()) {
+			testHarness.processAll();
```

Comment:
Why is this one needed here and not in the other test?

Reply:
Honestly it is not needed here. I just thought this way it better resembles the "real" process, where you processAll records and then finish the stream task.

In the second case the processAll could be misleading as one could say that after processingAll records it might have been finished already and cancelling makes no sense.

To sum up. It does not make a real difference if we have the `processAll` or not, but I thought it slightly better describes what is happening in reality.

Reply:
If you find it misleading I can remove the `processAll`.